### PR TITLE
Issue #2820865: PaymentGatewayBase should implement PluginWithFormsInterface

### PR DESCRIPTION
**Pull request by duozersk**
PR for https://www.drupal.org/node/2820865

### CODE REVIEW

File: modules/payment/src/Plugin/Commerce/PaymentGateway/PaymentGatewayBase.php [hunk 1]
```diff
@@ -11,12 +11,16 @@
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
 use Drupal\Core\Plugin\PluginBase;
+use Drupal\Core\Plugin\PluginWithFormsInterface;
```
Comment (bojanz) [inline modules/payment/src/Plugin/Commerce/PaymentGateway/PaymentGatewayBase.php]: PaymentGatewayInterface already extends PluginWithFormsInterface, you don't need to add it here.

File: modules/payment/src/Plugin/Commerce/PaymentGateway/PaymentGatewayBase.php [hunk 2]
```diff
@@ -11,12 +11,16 @@
...
+use Drupal\Core\Plugin\PluginWithFormsTrait;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
  * Provides the base class for payment gateways.
  */
-abstract class PaymentGatewayBase extends PluginBase implements PaymentGatewayInterface, ContainerFactoryPluginInterface {
+abstract class PaymentGatewayBase extends PluginBase implements PaymentGatewayInterface, ContainerFactoryPluginInterface, PluginWithFormsInterface {
```
Comment (bojanz) [inline modules/payment/src/Plugin/Commerce/PaymentGateway/PaymentGatewayBase.php]: You also need to remove the methods that are now provided by the trait.

**Review by bojanz — CHANGES_REQUESTED** [review]

File: modules/payment/src/Plugin/Commerce/PaymentGateway/PaymentGatewayBase.php (hunk 1 above)
Comment (duozersk) [reply modules/payment/src/Plugin/Commerce/PaymentGateway/PaymentGatewayBase.php]: Yep, correct, fixed

File: modules/payment/src/Plugin/Commerce/PaymentGateway/PaymentGatewayBase.php (hunk 2 above)
Comment (duozersk) [reply modules/payment/src/Plugin/Commerce/PaymentGateway/PaymentGatewayBase.php]: Good-good, removed.

**Comment by bojanz** [comment]
Thanks!

**Pull request merged**